cmd/reviewdog/doghouse_test.go

```diff
@@ -270,8 +177,19 @@ func TestPostResultSet_withReportURL(t *testing.T) {
 		default:
 			t.Errorf("unexpected req.Name: %s", req.Name)
 		}
-		return &doghouse.CheckResponse{ReportURL: "xxx"}, nil
-	}
+
+		if err := json.NewEncoder(w).Encode(&doghouse.CheckResponse{
+			ReportURL: "xxx",
+		}); err != nil {
+			t.Fatal(err)
+		}
+		w.WriteHeader(http.StatusOK)
+	})
+	ts := httptest.NewServer(mux)
+	defer ts.Close()
+
+	cli := client.New(&http.Client{})
```

Comment:
nit: we can use https://pkg.go.dev/net/http/httptest#Server.Client

```suggestion
	cli := client.New(ts.Client())
```